web/data/replication.go

```diff
@@ -79,14 +78,57 @@ func bulkGet(c echo.Context) error {
 	return proxy(c, "_bulk_get")
 }
 
+func bulkDocs(c echo.Context) error {
+	doctype := c.Get("doctype").(string)
+
+	if err := permissions.AllowWholeType(c, permissions.POST, doctype); err != nil {
+		return err
+	}
+
+	if err := CheckWritable(doctype); err != nil {
+		return err
+	}
+
+	instance := middlewares.GetInstance(c)
+	p, req, err := couchdb.ProxyBulkDocs(instance, doctype, c.Request())
+	if err != nil {
+		var code int
+		if errHTTP, ok := err.(*echo.HTTPError); ok {
+			code = errHTTP.Code
+		} else {
+			code = http.StatusInternalServerError
+		}
+		return c.JSON(code, echo.Map{
+			"error": err.Error(),
+		})
+	}
+
+	p.ServeHTTP(c.Response(), req)
+	return nil
+}
+
+func createDB(c echo.Context) error {
+	doctype := c.Get("doctype").(string)
+
+	if err := permissions.AllowWholeType(c, permissions.POST, doctype); err != nil {
+		return err
+	}
+
+	if err := CheckWritable(doctype); err != nil {
+		return err
+	}
+
+	return proxy(c, "/")
+}
+
 func fullCommit(c echo.Context) error {
 	doctype := c.Get("doctype").(string)
 
-	if err := permissions.AllowWholeType(c, permissions.GET, doctype); err != nil {
+	if err := permissions.AllowWholeType(c, permissions.POST, doctype); err != nil {
```

Comment:
`_ensure_full_commit` is not used for replication from couchdb to pouchdb (ie when we are just reading from couchdb)?